### ISSUE TITLE
Only One Client Receiving Updates

### ISSUE THREAD

**Issue by joelcdoyle** [dsuryd/dotNetify#58]
This is surely not a bug, but my lack of understanding...
When I start a simple CRUD app on `localhost:3000` and open two browser windows as clients, when I update a row in one client, I don't ever receive the new list of rows in the other client. Shouldn't the server broadcast the changes and each browser tab receive the updates?

view_models/EmployeesList.cs:
```
    public List<EmployeeModel> Employees => this.GetEmployeeList();  // reads from database
    public string Employees_itemKey => nameof(EmployeeModel.RecId);
    public Action<EmployeeModel> Update => employee =>
    {
      dataHelper.EmployeeUpdate(employee);  //writes to database
      this.UpdateList(nameof(Employees), employee);  //updates the list
      Changed(nameof(Employees));  //explicity mark as changed
      PushUpdates();  //force push of updates
    };
```

EmployeeList.jxs
```
import React from 'react';
import dotnetify from 'dotnetify';
dotnetify.hubServerUrl = "http://localhost:5000"; //fix for react-scrtips 1.0.16

class EmployeeList extends React.Component {
  constructor(props) {
    super(props);
    this.dispatch = state => this.vm.$dispatch(state);
    this.dispatchState = state => {
      this.setState(state);
      this.vm.$dispatch(state);
    }

    this.vm = dotnetify.react.connect("EmployeeList", this);
    this.state = { Employees: [] };

    this.updateStatus = this.updateStatus.bind(this);
  }

  updateStatus(value) {
    let updatedEmployee = {};
    let updatedList = this.state.Employees.map(employee => {
      if (employee.RecId === value.RecId) {
        updatedEmployee = Object.assign({}, employee, value);
      }
      return employee.RecId === value.RecId ? Object.assign({}, employee, value) : employee;
    });
    this.setState({ Employees: updatedList });
    this.dispatch({ Update: updatedEmployee });
  }

  componentWillUnmount() {
    this.vm.$destroy();
  }

  render() {
    let employees = this.state.Employees.map((emp) => {
      return (
        <Employee
          onStatusChange={this.updateStatus}
          key={emp.RecId.toString()}
          name={emp.Name} id={emp.RecId}
          status={emp.Status} />
      );
    });
    return employees;
  }
}
export default EmployeeList;

class Employee extends React.Component {
  constructor(props) {
    super(props);
    this.setIn = this.setIn.bind(this);
    this.setHome = this.setHome.bind(this);
    this.setAway = this.setAway.bind(this);
  }
  setIn() {
    this.props.onStatusChange({ RecId: this.props.id, Status: 0 });
  }
  setHome() {
    this.props.onStatusChange({ RecId: this.props.id, Status: 1 });
  }
  setAway() {
    this.props.onStatusChange({ RecId: this.props.id, Status: 2 });
  }

  render() {
    return (
      <div>
        {this.props.name} /
        {this.props.id} /
        {this.props.status}
        <button type="button" onClick={this.setIn}>Set In</button>
        <button type="button" onClick={this.setHome}>Set Home</button>
        <button type="button" onClick={this.setAway}>Set Away</button>
      </div>
    )
  }
}
```

Currently, only the browser window I am interacting with updates, and not the background window. The only way to see the updates in the second client is to refresh the page.

**Comment by dsuryd** [comment]
View model instances have 1-to-1 relationship with their views; not meant to fan out.  You need a backing pub/sub service injected into the view models for that.   Having built-in event aggregator is in the roadmap.

**Comment by joelcdoyle** [comment]
Thank you!

**Comment by joshua1** [comment]
Wait, This is the same View (different instance though) in a different browser. I thought Changed and PushChanges is supposed to notify/update connected clients

**Comment by joshua1** [comment]
Can changed and PushChanges be extended to notify other running instances , that way other clients can be notified from their server instance

**Comment by dsuryd** [comment]
Reopening this issue.  The use case to be addressed is synchronizing updates to clients that share the same view model.  My original thought is to put the responsibility on the developer to explicitly program this behavior through abstractions like an event aggregator, but will consider other approaches where the framework can be configured to take care of this in certain scenarios.    

Cross-reference: [dotnetify-react-demo-vs-2017/issue #7](https://github.com/dsuryd/dotnetify-react-demo-vs2017/issues/7)

**Comment by Caldas** [comment]
Good idea to reopen, I also believe that users could easy set 1-1 pra 1-N behavior at the view

**Comment by koo9** [comment]
it maybe easier with .net core 2.1 as signalR will be there. pub/sub is builtin.

**Comment by dsuryd** [comment]
This issue has been addressed in Nuget v3.1 by the introduction of new base class __MulticastVM__.   
Read the documentation here:  http://www.dotnetify.net/core/api/multicast
